SpaceX/HistoryDetailsScreen/HistoryDetailViewController.swift

```diff
@@ -6,16 +6,206 @@
+                         size: CGSize(width: 0, height: 250))
+    }
+    
+    private func setUoMissionNameLabel() {
+        missionNameLabel.numberOfLines = 0
+        let missionNameAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont(name: "Inter-Regular", size: 15) as Any
+        ]
+        let nameAttributes: [NSAttributedString.Key: Any] = [
+            .font: UIFont(name: "Inter-Regular", size: 18) as Any,
+            .foregroundColor: UIColor.sxRed
+        ]
+        
+        let missionNameString = NSMutableAttributedString(string: "Mission name ", attributes: missionNameAttributes)
+        
+        let name = NSAttributedString(string: flight().missionName ?? "", attributes: nameAttributes)
+        
+        missionNameString.append(name)
+        missionNameLabel.attributedText = missionNameString
+    }
+    
+    private func setUpInfoView() {
+        let infoViewElements = InfoView(flight: flight())
+        infoView.addSubview(infoViewElements)
+        infoViewElements.anchor(top: infoView.topAnchor,
+                                leading: infoView.leadingAnchor,
+                                bottom: infoView.bottomAnchor,
+                                trailing: infoView.trailingAnchor)
+    }
+    
+    private func setUpVideoView() {
+        videoView.layer.masksToBounds = true
+        videoView.layer.cornerRadius = 10
+       
+        guard let url = URL(string: data?.flight?.fragments.apIflight.links?.videoLink ?? "") else { return }
+        do {
+            try self.videoView.loadVideo(withUrl: url, playerVars: .init(playsInline: .true))
+        } catch {
+            print(error)
+        }
+    }
+    
+    
+    private func createRocketView() {
+        let rocketView = RocketView(flight: flight().rocket?.rocket)
+        scrollView.addSubview(rocketView)
+        rocketView.anchor(top: videoView.bottomAnchor,
+                          leading: scrollView.contentLayoutGuide.leadingAnchor,
+                          bottom: scrollView.contentLayoutGuide.bottomAnchor,
+                          trailing: scrollView.contentLayoutGuide.trailingAnchor,
+                          spacing: .init(top: 15, left: 20, bottom: 15, right: 20))
+    }
+    
+    private func showInfoNotFound() {
+        let infoNotFound = UILabel()
+        view.addSubview(infoNotFound)
+        infoNotFound.text = "Sorry, info not found 😔"
+        infoNotFound.font = UIFont(name: "Inter-Regular", size: 15)
+        
+        infoNotFound.translatesAutoresizingMaskIntoConstraints = false
+        infoNotFound.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor).isActive = true
+        infoNotFound.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor).isActive = true
+    }
+    
+    private func flight() -> ApIflight {
+        guard let flight = data?.flight?.fragments.apIflight  else { return ApIflight() }
+        return flight
     }
```

Comment:
we can add private `var flight` on top level
we will be similar as `var data`
and in `init` we can pass data to `var flight`
init will be
```
init(data: HomeInfoQuery.Data.History?) {
         self.data = data
         self.flight = data?.flight?.fragments.apIflight
         super.init(nibName: nil, bundle: nil)
     }
```